examples/scala-parallel-similarproduct/filterbyyear/src/main/scala/DataSource.scala

```diff
@@ -29,7 +29,8 @@ class DataSource(val dsp: DataSourceParams)
 
   override
   def readTraining(sc: SparkContext): TrainingData = {
-    val eventsDb = cleanedPEvents(sc)
+    cleanAndPersistPEvents(sc)
+    val eventsDb = Storage.getPEvents()
 
```

Comment:
If it's more efficient can't we still return the events from cleanAndPersistPEvents, instead of dropping them then reading them again with Storage.getPEvents()?